substrate/frame/support/test/tests/pallet.rs

```diff
@@ -2381,8 +2430,10 @@ fn post_runtime_upgrade_detects_storage_version_issues() {
 	>;
 
 	TestExternalities::default().execute_with(|| {
-		// Call `on_genesis` to put the storage version of `Example` into the storage.
-		Example::on_genesis();
```

Comment:
This test used to rely on `Example2` pallet not initialising its storage version, causing the below on-chain version <-> current version check to fail.

Updated this test to work with the new system where *initialisation* is automatic, but once the version is initialised the dev will need to apply migrations to keep on-chain and current versions consistent.